src/misc_gui.cpp

```diff
@@ -794,7 +794,7 @@ void QueryString::DrawEditBox(const Window *w, WidgetID wid) const
 
 	DrawFrameRect(cr, wi->colour, wi->IsLowered() ? FR_LOWERED : FR_NONE);
 	DrawSpriteIgnorePadding(rtl ? SPR_IMG_DELETE_RIGHT : SPR_IMG_DELETE_LEFT, PAL_NONE, cr, SA_CENTER);
-	if (this->text.bytes == 1) GfxFillRect(cr.Shrink(WidgetDimensions::scaled.bevel), _colour_gradient[wi->colour & 0xF][2], FILLRECT_CHECKER);
```

Comment:
What was this `colour & 0xF` doing? (here and elsewhere)

Reply:
It was making sure the value of `wi->colour` is within bounds (not always done) which is now done by `GetColourGradient()`

Reply:
Ah, thanks.